src/test/java/io/neonbee/test/helper/MultipartResponse.java

```diff
@@ -0,0 +1,182 @@
+     */
+    @VisibleForTesting
+    static MultipartResponse of(HttpResponse<Buffer> batchResponse) {
+        String boundary = extractBoundary(batchResponse);
+        String delimiter = "--" + boundary;
+        String body = batchResponse.body().toString();
+        ArrayList<Part> responseParts = new ArrayList<>();
+
+        int startIndex = 0;
+        while ((startIndex = body.indexOf(delimiter, startIndex)) >= 0) {
+            int nextIndex = body.indexOf(delimiter, startIndex + 1);
+            if (nextIndex < 0) {
+                // this can only happen on the final delimiter line
+                break;
+            }
+
+            int endOfLineIndex = body.indexOf('\n', startIndex);
+            String partString = body.substring(endOfLineIndex + 1, nextIndex - 1);
+            responseParts.add(parseAsPart(partString));
+            startIndex = nextIndex;
+        }
+
+        return new MultipartResponse(boundary, responseParts);
+    }
+
+    @VisibleForTesting
+    static String extractBoundary(HttpResponse<Buffer> batchResponse) {
+        String contentType = Optional.ofNullable(batchResponse.headers().get(CONTENT_TYPE)).orElseThrow(
+                () -> new IllegalArgumentException("batchResponse does not contain mandatory header " + CONTENT_TYPE));
+        int indexOfBoundary = contentType.indexOf("boundary");
+        if (indexOfBoundary < 0) {
+            throw new IllegalArgumentException("batchResponse header " + CONTENT_TYPE + " does not contain boundary!");
+        }
+        String[] boundaryKeyAndValue = contentType.substring(indexOfBoundary).split("=", 2);
+        if (boundaryKeyAndValue.length < 2) {
+            throw new IllegalArgumentException(
+                    "Boundary not properly specified in batchResponse header " + CONTENT_TYPE + "!");
+        }
+        return boundaryKeyAndValue[1];
+    }
+
+    /**
+     * Parses a string expected to be a part of an <a href=
+     * "http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_MultipartBatchResponse">OData
+     * Multipart Batch Response</a>. The string MUST NOT include the boundary delimiter.
+     *
+     * @param value string value of a batch response part
+     * @return part information extracted from given value string
+     */
+    @VisibleForTesting
+    static Part parseAsPart(String value) {
+        Collector<Map.Entry<String, String>, ?, Map<String, String>> headerCollector =
+                Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue);
+        Function<String, Map.Entry<String, String>> headerMapper = header -> {
+            String[] keyAndValue = header.split(":", 2);
+            return Map.entry(keyAndValue[0].trim(), keyAndValue[1].replaceFirst("\\R{1,2}$", "").trim());
+        };
+
+        String[] sections = value.split("\\R{2,}", 3);
+
+        Map<String, String> partHeaders = sections[0].lines().map(headerMapper).collect(headerCollector);
```

Comment:
Is this not always "Content-Type: application/http" except you have a change set what is not yet supported?

Reply:
As discussed, the part headers will net be parsed as they currently do not provide relevant information